jbi/runner.py

```diff
@@ -21,12 +24,71 @@
 
 logger = logging.getLogger(__name__)
 
+settings = get_settings()
+
+
+GROUP_TO_OPERATION = {
+    "new": Operation.CREATE,
+    "existing": Operation.UPDATE,
+    "comment": Operation.COMMENT,
+}
+
+
+def groups2operation(steps: ActionSteps):
+    """In the configuration files, the steps are grouped by `new`, `existing`,
+    and `comment`. Internally, this correspond to enums of `Operation`.
+    This helper remaps the list of steps.
+    """
+    try:
+        by_operation = {
+            GROUP_TO_OPERATION[entry]: steps_list
+            for entry, steps_list in steps.dict().items()
+        }
+    except KeyError as err:
+        raise ValueError(f"Unsupported entry in `steps`: {err}") from err
+    return by_operation
+
+
+class Executor:
+    """Callable class that encapsulates the default action."""
```

Comment:
Shall we mention *default* here? Or just call it *encapsultes the workflow*

Reply:
[af68f63](https://github.com/mozilla/jira-bugzilla-integration/pull/591/commits/af68f6336ef2b60e5d0773f4d880acfa6cc84fb4)